### DISCUSSION TITLE
P3 point

### DISCUSSION THREAD

**Discussion by adrimg89** [cwapi3d/cwapi3dpython#223] [Q&A]
hello everybody!

Does anyone know how p3 point works?

I always have problems with it.

Teorically, p3 point is  normal vector's direction for the reference face. Therefore, if my p3=cadwork.point_3D(0,0,1) when I create a beam my reference face would be perpendicular to z axis. 

Is that ok?

When I put that in practice, it doesn't work like that, and my beam is created in a strange position.

Thank you very much in advance

**Comment by Antoine-S-C** [comment]
I think p3 should be in global coordinates. It is the direction of the height axis of the element.
Example: if your element is a beam oriented along X and placed at a height of one meter (1000mm), its p3 should be (0,0,1001) or (0,0,2000). With p3(0,0,1) it would be upside down.

**Reply by adrimg89** [reply]
Thank you, Antoine. I think I am starting to understand it. 
Anyway, if I create any beam with create_rectangular_beam_VECTORS method, it works perfectly, so it's enough for my diferents codes.

**Comment by Brunner246** [comment]
Hi, 

P3 (point3) is the vector (normalized) for the orientation of the elements local z-axis. 

![image](https://github.com/user-attachments/assets/1800b034-e1d5-46ff-8dea-c73abccb79bf)

![image](https://github.com/user-attachments/assets/249c03bc-a5c8-435d-97c3-c1a9e79103d0)



```python
import cadwork
import element_controller as ec

axis_start_point = cadwork.point_3d(0.,0.,0.)
axis_end_point = cadwork.point_3d(1200.,0.,0.)

axis_z_direction = cadwork.point_3d(0., -1., 0.) # orientation for point 3

ec.create_rectangular_beam_points(100., 200., axis_start_point,
                                  axis_end_point,
                                  axis_z_direction)
1547742446
import geometry_controller as gc
gc.get_p3(1547742446)
[0.000000, -1.000000, 0.000000]
```

**Reply by adrimg89** [reply]
Hello @Brunner246 !

I think that is not exactly like you tell me. 

The vector you are defining is zl, not P3. 

Here you have an image:

![image](https://github.com/user-attachments/assets/3ac87086-f516-47c6-8b43-3fc1ad52ac1b)

**Reply by Brunner246** [reply]
P3 is placed where the local z axis of an element points. 

At least two vectors must be known in order to create the local axis system. In the above example, the x axis is defined via P1 and P2, P3 then defines the local z axis, axis y is derived.

![image](https://github.com/user-attachments/assets/a685f20f-9649-483c-a5e9-90d81ef4ed0a)

**Reply by adrimg89** [reply]
ok! thanks!!
